mongodb/client.bal

```diff
@@ -21,51 +21,188 @@ import ballerina/jballerina.java;
+    # + return - An array of database names on success or else a `mongodb:DatabaseError` if unable to reach the DB 
     remote function getDatabasesNames() returns string[]|DatabaseError {
         return getDatabasesNames(self.datasource);
     }
 
-    # Returns the `Database` client.
-    # ```ballerina
-    # mongodb:Database|mongodb:Error result = mongoClient->getDatabase("Ballerina");
-    # ```
+    # Returns the `Database` handle.
     # 
-    # + name - Name of the database
-    # + return - A database client object on success or else a `mongodb:Error` if unable to reach the DB
-    remote function getDatabase(string name) returns Database|Error {
-        if (name.trim().length() == 0) {
+    # + databaseName - Name of the database
+    # + return - A database handle on success or else a `mongodb:Error` if unable to reach the DB
+    function getDatabase(string databaseName) returns handle|Error {
+        if (databaseName.trim().length() == 0) {
             return error ApplicationError("Database Name cannot be empty.");
         }
+        handle database = check getDatabase(self.datasource, databaseName);
+        return database;
+    }
+
+    //Collection management operations 
+    # Lists the collection names in the MongoDB database.
+    #
+    # + databaseName - Name of the database 
+    # + return - An array of collection names on success or else a `mongodb:Error` if unable to reach the DB
+    remote function getCollectionNames(string? databaseName = ()) returns string[]|Error {
+        handle database = check self.getCurrentDatabase(databaseName);        
+        return getCollectionNames(database); 
+    }
+
+    # Returns the collection handle.
+    # 
+    # + databaseName - Name of the database 
+    # + collectionName - Name of the collection
+    # + return - A collection object on success or else a `mongodb:Error` if unable to reach the DB
+    function getCollection(string collectionName, string? databaseName = ()) returns handle|Error {
+        if (collectionName.trim().length() == 0) {
+            return error ApplicationError("Collection Name cannot be empty.");
+        }
+        handle database = check self.getCurrentDatabase(databaseName);
+        handle collection = check getCollection(database, collectionName);
+        return collection;
+    }
+
+    // Collection service operations
+    # Counts the documents based on the filter. When the filter is (), it counts all the documents in the collection.
+    #
+    # + databaseName - Name of the database 
+    # + collectionName - Name of the collection
+    # + filter - Filter for the count ($where & $near can be used)
+    # + return - Count of the documents in the collection or else `mongodb:Error` if unable to reach the DB
+    remote function countDocuments(string collectionName, string? databaseName = (), map<json>? filter = ()) 
+    returns int|Error {
```

Comment:
Bring the return statement inline with the staring bracket of the method signature